### PR TITLE
feat(x10forge)

### PR DESCRIPTION
**Pull request by Nyawmeowww**
### All Submissions

image
image

### New Feature Submissions

Issue #619  and Issue #628

### Notes

Please note, your code must pass all tests and lint checks before it can be merged.

### PR Description

The fixed in this PR is only for the modal not showing any weapons. The contract fix for Issue #619 is still in progress

### CODE REVIEW

File: frontend/src/views/Blacksmith.vue [hunk 1]
```diff
@@ -246,13 +246,20 @@ export default {
...
       } finally {
+        this.disableForge = false;
+      }
+
+      try{
+        this.viewNewWeapons(1);
+      } catch (e) {
+        console.error(e);
+        this.onError = true;
+        this.$dialog.notify.error('This one belongs to viewNewWeapons x1');
```
Comment (seiyria) [inline frontend/src/views/Blacksmith.vue:261]: this is debugging stuff, remove it

File: frontend/src/views/Blacksmith.vue [hunk 2]
```diff
@@ -274,13 +281,20 @@ export default {
...
       } finally {
+        this.disableForge = false;
+      }
+
+      try{
+        this.viewNewWeapons(this.forgeMultiplier);
+      } catch (e) {
+        console.error(e);
+        this.onError = true;
+        this.$dialog.notify.error('This one belongs to viewNewWeapons x10');
```
Comment (seiyria) [inline frontend/src/views/Blacksmith.vue:296]: debugging

File: frontend/src/views/Blacksmith.vue [hunk 3]
```diff
@@ -311,8 +325,17 @@ export default {
...
     },
 
-    viewNewWeapons(newWeaponCount = 1){
-      this.newForged = this.ownedWeaponIds.slice(-newWeaponCount);
+    viewNewWeapons(){
+      this.newForged = [];
+      this.ownedWeaponIds.forEach(x => {
+        this.newForged.push(x);
+      });
+      if(this.x1Forge === true){
```
Comment (seiyria) [inline frontend/src/views/Blacksmith.vue:333]: never check against true, just check if(this.x1Forge). perhaps instead, do `this.forgeAmount = 1 or 10` and check that. but really, this should be a number and you should do the math based on that number.

File: frontend/src/views/Blacksmith.vue [hunk 4]
```diff
@@ -330,10 +326,10 @@ export default {
       this.ownedWeaponIds.forEach(x => {
         this.newForged.push(x);
       });
-      if(this.x1Forge === true){
+      if(this.forgeMultiplier === 1){
```
Comment (seiyria) [inline frontend/src/views/Blacksmith.vue:329]: why do this - use forgeMultiplier to do the math for splicing

**Review by seiyria — APPROVED** [review]

File: frontend/src/views/Blacksmith.vue [hunk 5]
```diff
@@ -305,14 +304,16 @@ export default {
...
+      this.newForged = [];
       this.ownedWeaponIds.forEach(x => {
-        this.currentListofWeapons.push(x);
+        this.newForged.push(x);
       });
-    },
 
-    viewNewWeapons(newWeaponCount = 1){
-      this.newForged = this.ownedWeaponIds.slice(-newWeaponCount);
+      ++offset;
```
Comment (seiyria) [inline frontend/src/views/Blacksmith.vue:313]: this really holds no purpose. I will fix and merge

**Review by seiyria — APPROVED** [review]

**Comment by seiyria** [comment]
also, in the future, branch off of your fork. we're trying to keep this repo mostly clean

**Comment by Nyawmeowww** [comment]
Okii

**Pull request merged**